File: examportal/src/main/java/com/exam/service/QuizService.java
package com.exam.service;

import java.util.List;
import java.util.Set;

import com.exam.entity.exam.Category;
import com.exam.entity.exam.Quiz;

public interface QuizService {
	
	public Quiz addQuiz(Quiz quiz);
	public Quiz updateQuizz(Quiz quiz);
	public Set<Quiz> getQuizzes();
	public Quiz getQuiz(Long qId);
	public void deleteQuiz(Long qId);
	public List<Quiz> getQuizOfCategory(Category category);
	public List<Quiz> getActiveQuizzes();
	public List<Quiz> getActiveQuizzesOfCategory(Category c);
}

File: examfront/src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private http:HttpClient) { }

  public getQuizzes()
  {
    return this.http.get(`${baseUrl}/quiz/`);
  }

  public getQuiz(qid:any)
  {
    return this.http.get(`${baseUrl}/quiz/${qid}`);
  }

  public deleteQuiz(qid: any) {
    return this.http.delete(`${baseUrl}/quiz/${qid}`); 
  }

  public addQuiz(quiz:any)
  {
    return this.http.post(`${baseUrl}/quiz/`,quiz);
  }

  public updateQuiz(quiz:any)
  {
    return this.http.put(`${baseUrl}/quiz/`,quiz);
  }

  public getQuizOfCategory(cid:any)
  {
    return this.http.get<any[]>(`${baseUrl}/quiz/category/${cid}`)
  }

  // get active quizzes
  public getActiveQuizzes()
  {
    return this.http.get<any[]>(`${baseUrl}/quiz/active`)
  }

  // get active quizzes of category
  public getActiveQuizzesOfCategory(cid:any)
  {
    return this.http.get<any[]>(`${baseUrl}/quiz/active/category/${cid}`)
  }

}
